Userland/Libraries/LibGUI/TextDocument.h

```diff
@@ -246,6 +246,29 @@ class RemoveTextCommand : public TextDocumentUndoCommand {
     TextRange m_range;
 };
 
+class InsertLineCommand : public TextDocumentUndoCommand {
+public:
+    enum class InsertPosition {
+        Above,
+        Below,
+    };
+
+public:
+    InsertLineCommand(TextDocument&, TextPosition, String&&, InsertPosition pos);
+    virtual ~InsertLineCommand() = default;
+    virtual void undo() override;
+    virtual void redo() override;
+    virtual String action_text() const override;
+
+private:
+    size_t compute_line_number() const;
+
+private:
+    TextPosition m_cursor;
+    String m_text;
+    InsertPosition m_pos;
```

Comment:
Please don't mind the noise, I was just looking at this PR and I was wondering, is there a reason why two `private` and `public` blocks? Probably a design practice in Serenity, that I'm unaware of. Was just curious :)

Reply:
Well, actuallt they are redundant, it's just my personal "coding style" to divide things of class into different groups, like private inner functions and private members...
I think it's better for huge classes and makes code more readable, but use a single `private` and `public` here will be better :^)

Reply:
Got it, thank you so much for answering my query 💯